src/gt4py/next/iterator/embedded.py

```diff
@@ -887,20 +869,26 @@ def make_in_iterator(
             sparse_dimensions.append(axis.value)
 
     new_pos: Position = pos.copy()
-    for sparse_dim in set(sparse_dimensions):
-        init = [None] * sparse_dimensions.count(sparse_dim)
+    sparse_dimensions = list(set(sparse_dimensions))
+    assert len(sparse_dimensions) <= 1, sparse_dimensions
+    for sparse_dim in sparse_dimensions:
+        init = [None]  # * sparse_dimensions.count(sparse_dim)
         new_pos[sparse_dim] = init  # type: ignore[assignment] # looks like mypy is confused
     if column_axis is not None:
         # if we deal with column stencil the column position is just an offset by which the whole column needs to be shifted
         assert _column_range is not None
         new_pos[column_axis] = _column_range.start
-    return MDIterator(
+    it = MDIterator(
         inp,
         new_pos,
-        incomplete_offsets=[SparseTag(x) for x in sparse_dimensions],
+        incomplete_offsets=[],
```

Comment:
Are `incomplete_offsets` still used in actual code? If they are useless and meant to be removed soon, I would add a TODO comment in the `MDIterator` class definition.

Reply:
added a comment